queue.tf

```diff
@@ -23,13 +27,34 @@ module "subscription" {
   source    = "./subscription"
   topic_arn = join("",
     [
-      "arn:aws:sns:${data.aws_region.current.name}",
-      ":${data.aws_caller_identity.current.account_id}:",
-      each.value.name
+      can(regex("^arn:aws:sns", each.value.name)) ?
+      each.value.name :
+      "arn:aws:sns:${data.aws_region.current.name}:${data.aws_caller_identity.current.account_id}:${each.value.name}"
     ]
   )
   queue_name           = aws_sqs_queue.queue.name
   raw_message_delivery = each.value.raw_message_delivery
   depends_on           = [aws_sqs_queue.queue]
+  filter_policy = var.filter_policy
 }
 
+resource "aws_cloudwatch_metric_alarm" "create_order_dlq_error" {
+  alarm_name = "${aws_sqs_queue.dlq.name}_alarm"
+  actions_enabled = var.environment == "production" ? true: false
+  alarm_description = "Dead letter QUEUE should be empty"
+  tags = var.tags
+  namespace = "AWS/SQS"
+  metric_name = "ApproximateNumberOfMessagesVisible"
+  statistic = "Maximum"
+  comparison_operator = "GreaterThanThreshold"
+  threshold = 0
+  period = 60
+  dimensions = {
+    QueueName = aws_sqs_queue.dlq.name
+  }
+  alarm_actions = [
+    aws_sns_topic.cloud_watch_alarms.arn
```

Comment:
esse recurso está definido?, se não, esse arn deve ser um input

Reply:
Tá sim. Tá na [linha 1](https://github.com/dev-clubbi/terraform-sns-sqs-with-dlq/pull/1/files#diff-9701154d5cc5faccc340851bc07333705643e22604b414df07d27eaecf37fe29R1):
```
resource "aws_sns_topic" "cloud_watch_alarms" {
  name = "${var.prefix}alerts_ops"
  tags = var.tags
}
```

Reply:
Hmmm, acho que seria melhor passar o nome do cloud_watch_alarm ou o arn dele não? pq outros times podem querer usar outros tópicos, essa mudança iria forçar todos alertas irem para o alerts_ops